app/views/social_accounts/_form.html.erb

```diff
@@ -0,0 +1,31 @@
+<%= form_with(model: [@user, social_account], local: true) do |form| %>
+  <% if self.controller.is_a?(SocialAccountsController) %>
```

Comment:
We are creating a *social account* in the `social account controller` but we are sending the request from the `user controller`. We are doing this since we want to add a social account from the *edit my profile page*. If the account was created unsuccessfully we want to render the user's edit view but still be in the processing of the current request so that the user can correct her/his input. The problem is now that we need to render the error message. That's we are needing this workaround.

Reply:
I can have a look later if there is a more elegant solution.

Reply:
Correct me if I am wrong but `<% if social_account.errors.any? %>` ensures that only the relevant errors are displayed here?

Reply:
Ah, I think I fixed it without knowing it. Since `social_account ` is now an instance variable we do not have the problem anymore 🙈 .